services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/crt/S3CrtAsyncHttpClientTest.java

```diff
@@ -287,7 +287,8 @@ public void operationWithResponseAlgorithms_optOutValidationFromClient_shouldHon
 
         s3NativeClientConfiguration = S3NativeClientConfiguration.builder()
                                                                  .endpointOverride(DEFAULT_ENDPOINT)
-                                                                 .credentialsProvider(null)
```

Comment:
Should we add a test case such that when we pass it as "null" it will use `DefaultCredentialsProvider.builder().build()` or a non null Provider and [not a singleton instance if its fine]

Reply:
I moved the default credentials provider assignment to DefualtS3CrtAsyncClient, so it's always present in S3NativeClientConfiguration. The reason I did that is to make sure `s3Client.serviceClientConfiguration().credentialsProvider()` returns the same instance as the one created by the SDK.

Sure, I can add a test assertion in DefaultS3CrtAsyncClientTest